File: simple-bank-system.ts
class Bank {
    balance: number[];

    constructor(balance: number[]) {
        this.balance = balance;
    }

    transfer(account1, account2, money) {
        if (this.balance.length < account1 || this.balance.length < account2)
            return false;
        
        if (this.balance[account1 - 1] - money < 0)
            return false;

        this.balance[account2 - 1] += money;
        this.balance[account1 - 1] -= money;
        
        return true;

    };

    deposit(account, money) {
        if (this.balance.length < account)
            return false;

        this.balance[account - 1] += money;
        
        return true;
    };

    withdraw(account, money) {
        const diff = this.balance[account - 1] - money;
        
        if (isNaN(diff) || diff < 0)
            return false;
        
        this.balance[account - 1] = diff;
        
        return true;
    };

}


const bank = new Bank([10, 100, 20, 50, 30])

console.log(bank.withdraw(10, 10))

File: simple-bank-system.rs
struct Bank(Vec<i64>);
impl Bank {
    fn new(b: Vec<i64>) -> Self {
        Self(b)
    }
    fn transfer(&mut self, a1: i32, a2: i32, m: i64) -> bool {
        a2 as usize <= self.0.len() && self.withdraw(a1, m) && self.deposit(a2, m)
    }
    fn deposit(&mut self, a: i32, m: i64) -> bool {
        let a = a as usize - 1;
        a < self.0.len() && {
            self.0[a] += m;
            true
        }
    }
    fn withdraw(&mut self, a: i32, m: i64) -> bool {
        let a = a as usize - 1;
        a < self.0.len() && self.0[a] >= m && {
            self.0[a] -= m;
            true
        }
    }
}
